packages/reason-relay/reason-relay-ppx/library/Mutation.re

```diff
@@ -0,0 +1,451 @@
+            unit
+          ) =>
+          ReasonRelay.Disposable.t =
+          (
+            ~environment: ReasonRelay.Environment.t,
+            ~variables: [%t typeFromGeneratedModule(["Types", "variables"])],
+            ~optimisticUpdater=?,
+            ~optimisticResponse=?,
+            ~updater=?,
+            ~onCompleted=?,
+            ~onError=?,
+            (),
+          ) => (
+            Internal.internal_commitMutation(
+              environment,
+              {
+                variables:
+                  variables
+                  ->[%e
+                      valFromGeneratedModule(["Internal", "convertVariables"])
+                    ]
+                  ->ReasonRelay_Internal.internal_cleanVariablesRaw,
+                mutation: [%e valFromGeneratedModule(["node"])],
+                onCompleted:
+                  Some(
+                    (res, err) =>
+                      switch (onCompleted) {
+                      | Some(cb) =>
+                        cb(
+                          res->[%e
+                                 valFromGeneratedModule([
+                                   "Internal",
+                                   "convertResponse",
+                                 ])
+                               ],
+                          Js.Nullable.toOption(err),
+                        )
+                      | None => ()
+                      },
+                  ),
+                onError:
+                  Some(
+                    err =>
+                      switch (onError) {
+                      | Some(cb) => cb(Js.Nullable.toOption(err))
+                      | None => ()
+                      },
+                  ),
+                optimisticResponse:
+                  switch (optimisticResponse) {
+                  | None => None
+                  | Some(r) =>
+                    Some(
+                      r->[%e
+                           valFromGeneratedModule([
+                             "Internal",
+                             "convertWrapRawResponse",
+                           ])
+                         ],
+                    )
+                  },
+                optimisticUpdater,
+                updater:
+                  switch (updater) {
+                  | None => None
+                  | Some(updater) =>
+                    Some(
+                      (store, r) =>
+                        updater(
+                          store,
+                          r->[%e
+                               valFromGeneratedModule([
+                                 "Internal",
+                                 "convertResponse",
+                               ])
+                             ],
+                        ),
+                    )
+                  },
+              },
+            ): ReasonRelay.Disposable.t
+          )
+      ],
+      [%stri
+        let commitMutationPromised:
```

Comment:
I don't think this method should be generated since relay does not support converting the mutation response to a promise. There is no way for the user to clean up the subscription.

Reply:
There will be no subscription to clean up with a mutation, as it's a one shot thing that'll complete itself. Unless I'm misunderstanding you?

Reply:
I am probably just misunderstanding RelayObservable, I thought that the caller always had call `dispose`.